### PR TITLE
Write test cases for Login Module

### PR DESCRIPTION
**Pull request by Shrawak**
Ticket Link:
https://app.clickup.com/t/864eue3xt

Description:
Test the login module
* Test for fill in form inputs and submits successfully and renders to the / route
* Test for rendering to the / route after successful login by employee
* Test for rendering to the /onboarding route if employee logs in for the first time
* Test error message is displayed or not when login fails and disapper after few seconds

Screenshots/GIFs:

image

image

image

image

Deployment Strategy:
-- yarn test

### CODE REVIEW

File: src/containers/Login/test/login.test.tsx [hunk 1]
```diff
@@ -34,39 +36,46 @@ const loginAction = jest
...
-    expect(submitButton).toBeInTheDocument();
-  });
-  test('fills in form inputs and submits successfully', async () => {
-    // authMockData();
+  // Fill in form inputs
+  await userEvent.type(emailInput, email);
+  await userEvent.type(passwordInput, password);
+
+  // Submit the form
+  fireEvent.click(submitButton);
```
Comment (duwalanise) [inline src/containers/Login/test/login.test.tsx:56]: we can use userEvent here as well

File: src/containers/Login/test/login.test.tsx [hunk 2]
```diff
@@ -80,4 +89,55 @@ describe('Login', () => {
...
+    fireEvent.click(submitButton);
+
+    // Check if the error message is displayed
+    expect(screen.getByRole('alert')).toHaveClass('MuiAlert-standardError');
+    expect(
+      screen.getByText(/email or password is incorrect/i)
+    ).toBeInTheDocument();
+
+    // Wait for the error message to disappear and wrapping the transition inside a test in act(...).
+    await act(async () => {
```
Comment (duwalanise) [inline src/containers/Login/test/login.test.tsx:126]: what does this code do?

File: src/containers/Login/test/login.test.tsx (hunk 1 above)
Comment (Shrawak) [reply src/containers/Login/test/login.test.tsx:56]: okay sure

File: src/containers/Login/test/login.test.tsx (hunk 2 above)
Comment (Shrawak) [reply src/containers/Login/test/login.test.tsx:126]: As we are waiting for the error message to disapper there is the change in state so need to wrap the waiting Promise inside the act function.

**Pull request merged**